### PR TITLE
🚸 Do not enforce websites as default filter format and add filter hint

### LINKED ISSUE #2588: Respect format in canonical links
via @tharders: our `<link rel="canonical">` elements currently don't share a format GET parameter and therefore always fallback to *websites*.

This is wrong for documents that shouldn't actually be available for *websites*.

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Fixes #2588. A common way to come to amp.dev is through Google search. As our canonical links don't have a `?format` parameter we defaulted to `websites` which had weird UX effects: a search for `amp-story-grid-layer` would bring you to the correct page but with a hint saying:

>Important: this component does not support your currently selected format websites!

This PR enhances UX by using the first format the visited document is available in as default and additionally adds a more prominent hint to filter the page with `amp-user-notification`.

Bildschirmfoto 2019-12-23 um 15 21 24

### CODE REVIEW

File: frontend/templates/views/partials/format-filter-hint.j2 [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{% if doc.formats %}
+{% do doc.styles.addCssFile('css/components/molecules/format-filter-hint.css') %}
+
+{% do doc.amp_dependencies.add('amp-user-notification', '0.1') %}
+<amp-user-notification class="ap-m-format-filter-hint" id="format-filter-hint" layout="nodisplay">
+  <div class="ap-m-format-filter-hint-content">
+    <div class="ap-m-format-filter-hint-note">Select your format for a more streamlined experience</div>
```
Comment (sebastianbenz) [inline frontend/templates/views/partials/format-filter-hint.j2]: @CrystalOnScript PTAL at the wording

File: pages/content/amp-dev/documentation/guides-and-tutorials/start/test.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+$title: Format test
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/start/test.md]: Is this intended to be committed?

File: platform/lib/templates/SupportedFormatsExtension.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  run(context, args) {
+    const formats = args.formats || [];
+
+    // Check if the user did not select a specific format
+    // and if so overwrite the default format with a more fitting one
+    if (context.ctx.requestedFormat) {
+      return;
+    }
+
+    context.ctx.format = formats[0];
```
Comment (sebastianbenz) [inline platform/lib/templates/SupportedFormatsExtension.js]: What about the case when there are no formats given?

File: platform/lib/templates/SupportedFormatsExtension.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    return new nodes.CallExtension(this, 'run', args);
+  }
+
+  /**
+   * Tries to figure out if there is a fitting default
+   * format and overwrites the one set by the global context
+   */
+  run(context, args) {
+    const formats = args.formats || [];
```
Comment (sebastianbenz) [inline platform/lib/templates/SupportedFormatsExtension.js]: I don't quite understand why we need an extension for this?

Comment (matthiasrohmer) [reply platform/lib/templates/SupportedFormatsExtension.js]: We don't need one necessarily. An equivalent solution would be to put the logic into the template:

```
{% if doc.formats %}
[% if not requestedFormat %]
[% set supportedFormats = {{ doc.formats|tojson }} %]
[% set format = supportedFormats[0] || 'websites' %]
[% endif %]
{% endif %}
```

File: platform/lib/templates/SupportedFormatsExtension.js (hunk 3 above)
Comment (matthiasrohmer) [reply platform/lib/templates/SupportedFormatsExtension.js]: Templating handled an undefined format gracefully, but now explicitly falls back to websites again.

File: pages/content/amp-dev/documentation/guides-and-tutorials/start/test.md (hunk 2 above)
Comment (matthiasrohmer) [reply pages/content/amp-dev/documentation/guides-and-tutorials/start/test.md]: No, thanks!

File: frontend/templates/views/partials/format-filter-hint.j2 (hunk 1 above)
Comment (sebastianbenz) [reply frontend/templates/views/partials/format-filter-hint.j2]: ping

**Pull request merged**